lib/base-compiler.ts

```diff
@@ -599,10 +601,12 @@ export class BaseCompiler implements ICompiler {
         userOptions?: string[],
     ): string[] {
         let options = ['-g', '-o', this.filename(outputFilename)];
-        if (this.compiler.intelAsm && filters.intel && !filters.binary) {
+        if (this.compiler.intelAsm && filters.intel && !filters.binary && !filters.binaryobject) {
```

Comment:
Nitpick! But important, as this will be part of the public API. Everywhere else we use `camelCase` for things - do you think it would make more sense to call this `binaryObject`? That would match the documentation about "supportsBinaryObject"?

We also need to update the REST API documentation to cover this: `docs/API.md` (probably only needs a line or two of quick explanation)

Reply:
Will do!